iquip/apps/scheduler.py

```diff
@@ -212,17 +245,77 @@ def sizeHint(self, option: QStyleOptionViewItem, index: QModelIndex) -> QSize:
         return experimentView.sizeHint()
 
 
+class SchedulerPostWorker(QObject):
+    """Worker for posting a request to the proxy server, targeting the scheduler.
+
+    Signals:
+        done: The signal is emitted when the procedure of the worker is done.
+
+    Attributes:
+        mode: The type of command that is requested to the server.
+        rid: The run identifier value of the target experiment.
+    """
+    done = pyqtSignal()
+
+    def __init__(self, mode: str, rid: int):
+        """Extended.
+
+        Args:
+            mode: The type of command that is requested to the server.
+            rid: The run identifier value of the target experiment.
+        """
+        super().__init__()
+        self.mode = mode
+        self.rid = rid
+
+    def run(self):
+        """Overridden."""
+        basePath = "http://127.0.0.1:8000/experiment/"
+        requests.post(basePath + self.mode, params={"rid": self.rid}, timeout=10)
+        self.done.emit()
+
+
 class SchedulerApp(qiwis.BaseApp):
     """App for displaying the submitted experiment queue.
 
     Attributes:
         schedulerFrame: The frame that shows the submitted experiment queue.
+        menu: The QMenu instance to display menu when right-clicked.
+        actions: The dictionary of possible outcomes made by menu.
     """
 
     def __init__(self, name: str, parent: Optional[QObject] = None):
         """Extended."""
         super().__init__(name, parent=parent)
         self.schedulerFrame = SchedulerFrame()
+        self.schedulerFrame.queueView.rightButtonPressed.connect(self.displayMenu)
+        self.menu = QMenu(self.schedulerFrame)
+        # TODO(giwon2004) Remove icon space from the menu list (use menu.setStyleSheet)
+        self.actions = {
+            "edit": self.menu.addAction("Edit"),
+            "delete": self.menu.addAction("Delete"),
+            "terminate": self.menu.addAction("Request termination")
+        }
+
+    @pyqtSlot(QMouseEvent)
+    def displayMenu(self, event: QMouseEvent):
+        """Displays the menu pop-up.
+
+        Args:
+            event: Information holder containing the clicked position.
+        """
+        for i in range(self.schedulerFrame.model.rowCount()):
+            index = self.schedulerFrame.model.index(i)
+            if self.schedulerFrame.queueView.rectForIndex(index).contains(event.pos()):
+                rid = self.schedulerFrame.model.data(index).rid
+                action = self.menu.exec_(event.globalPos())
+                if action == self.actions["edit"]:
+                # TODO(giwon2004) Create an app for editing scannables.
+                    pass
+                elif action == self.actions["delete"]:
+                    _thread_with_worker(SchedulerPostWorker("delete", rid), self).start()
```

Comment:
Unless the `QThread` object isn't stored as an attribute of this class, does it work well?

I'm worried becuase the thread object may be removed before it finishes because the thread is defined only in `displayMenu()`.

Reply:
I have tested the thread object, and the termination of `displayMenu` did not delete it.
I guess it works well as the thread has a parent `self`.

However, I found a new problem: It does not finish well. It continues to run forever.

Reply:
> I have tested the thread object, and the termination of displayMenu did not delete it.
I guess it works well as the thread has a parent self.

Good!

Reply:
> However, I found a new problem: it does not run worker.run here! What could be the problem?

What do you mean?
Even if the thread starts, isn't `worker.run()` called?

Reply:
> Even if the thread starts, isn't worker.run() called?
No, it does not call the function.

Reply:
Umm.. I think the problem should be resolved in this PR..

Reply:
How did you check the problem?
In this branch, fetching the actual list from the proxy server isn't implemented yet, so I can't open the menu.

Reply:
I made a dummy experiment by manually adding an experiment by `self.addExperiment`. 
Then I used right-click on it and clicked Edit.

Here I made the thread have nothing to do (command-ize the `request` part), but the error message `QThread destroyed while running` showed up after I quit the app while it did not show when the function `_thread_with_worker` is not called.

This showed the thread does not end when the call itself dies, but also showed the thread does not end _until the end_.